p2p/session.go

```diff
@@ -195,21 +222,22 @@ func (s *session[H]) doRequest(
 		"requestedAmount", req.Amount,
 	)
 
+	remainingHeaders := req.Amount - uint64(len(h))
+
+	span.SetStatus(codes.Ok, fmt.Sprintf("request succeed. remaining header %d", remainingHeaders))
```

Comment:
From sources:
![image](https://github.com/celestiaorg/go-header/assets/22449364/ae2e1977-01e1-42f6-8aa8-7716294e6629)

So, description is ignored on success

Reply:
If we need to track remaining headers, the event would be perfect here. We can send the event after the if remainingHeader > 0 to signal there are more headers to fetch